### PR TITLE
In-page wallet prototype

### LINKED ISSUE #14: Prototype in-page wallet
PR: https://github.com/getwax/wax/pull/13

### PR DESCRIPTION
**Pull request by voltrevo**
Resolves #14.

Video walkthrough:

[Video walkthrough](https://www.youtube.com/watch?v=XuwlsWW_wo4)

### CODE REVIEW

File: inpage/package.json [hunk 1]
```diff
@@ -0,0 +1,49 @@
+{
+  "name": "@getwax/inpage",
+  "private": true,
+  "version": "0.1.0",
+  "type": "module",
+  "types": "build/lib/index.d.ts",
+  "scripts": {
+    "setup": "yarn && yarn --cwd hardhat && yarn --cwd hardhat hardhat compile",
+    "dev": "concurrently --kill-others vite \"yarn --cwd hardhat hardhat node\"",
```
Comment (blakecduncan) [inline inpage/package.json:9]: I really like this `concurrently` set up 👍

File: inpage/src/WaxInPage.tsx [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+import React from 'react';
+
+import EthereumApi from './EthereumApi';
+import assert from './helpers/assert';
+import popupUrl from './popupUrl';
+import PermissionPopup from './PermissionPopup';
+import sheetsRegistry from './sheetsRegistry';
+
+const defaultConfig = {
+  requirePermission: true,
```
Comment (blakecduncan) [inline inpage/src/WaxInPage.tsx:11]: I like the require permission config

File: inpage/src/sheetsRegistry.ts [hunk 3]
```diff
@@ -0,0 +1,6 @@
+import jss, { SheetsRegistry } from 'jss';
```
Comment (blakecduncan) [inline inpage/src/sheetsRegistry.ts:1]: Never heard of `jss` this is a cool package

File: inpage/src/EthereumApi.ts [hunk 4]
```diff
@@ -0,0 +1,101 @@
+import assert from './helpers/assert';
+import randomId from './helpers/randomId';
+
+export default class EthereumApi {
+  #networkUrl = 'http://127.0.0.1:8545';
+  #testAddress = '0xf39Fd6e51aad88F6F4ce6aB8827279cffFb92266';
```
Comment (blakecduncan) [inline inpage/src/EthereumApi.ts:6]: What do you think about how we will initialize this address when it's not hardcoded?  

For example, for an EOA do you think the user would be able to enter their pk in the popup?

File: inpage/demo/DemoContext.ts [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+import { createContext, useContext } from 'react';
+import WaxInPage, { EthereumApi } from '../src';
+
+export default class DemoContext {
+  ethereum: EthereumApi;
+  provider: ethers.BrowserProvider;
+
+  constructor(public waxInPage: WaxInPage) {
+    this.ethereum = waxInPage.ethereum;
+    this.provider = new ethers.BrowserProvider(this.ethereum);
```
Comment (blakecduncan) [inline inpage/demo/DemoContext.ts:12]: BrowserProvider 💯

**Review by blakecduncan — APPROVED** [review]
Nice!  I watched your demo video, and I think this is really cool.

File: inpage/src/EthereumApi.ts (hunk 4 above)
Comment (voltrevo) [reply inpage/src/EthereumApi.ts:6]: I think there should be a normal onboarding process, which a dapp could replace by directly assigning a private key / passphrase.

**Pull request merged**